llm/llama.go

```diff
@@ -33,46 +32,52 @@ type ModelRunner struct {
 }
 
 func chooseRunners(workDir, runnerType string) []ModelRunner {
-	buildPath := path.Join("llama.cpp", runnerType, "build")
+	buildPath := filepath.Join("llama.cpp", runnerType, "build")
 	var runners []string
 
 	// set the runners based on the OS
 	// IMPORTANT: the order of the runners in the array is the priority order
 	switch runtime.GOOS {
 	case "darwin":
 		runners = []string{
-			path.Join(buildPath, "metal", "bin", "server"),
-			path.Join(buildPath, "cpu", "bin", "server"),
+			filepath.Join(buildPath, "metal", "bin", "server"),
+			filepath.Join(buildPath, "cpu", "bin", "server"),
 		}
 	case "linux":
 		runners = []string{
-			path.Join(buildPath, "cuda", "bin", "server"),
-			path.Join(buildPath, "cpu", "bin", "server"),
+			filepath.Join(buildPath, "cuda", "bin", "server"),
+			filepath.Join(buildPath, "cpu", "bin", "server"),
 		}
 	case "windows":
 		// TODO: select windows GPU runner here when available
 		runners = []string{
-			path.Join(buildPath, "cpu", "bin", "Release", "server.exe"),
+			filepath.Join(buildPath, "cpu", "bin", "Release", "server.exe"),
 		}
 	default:
 		log.Printf("unknown OS, running on CPU: %s", runtime.GOOS)
 		runners = []string{
-			path.Join(buildPath, "cpu", "bin", "server"),
+			filepath.Join(buildPath, "cpu", "bin", "server"),
 		}
 	}
 
 	runnerAvailable := false // if no runner files are found in the embed, this flag will cause a fast fail
 	for _, r := range runners {
 		// find all the files in the runner's bin directory
-		files, err := fs.Glob(llamaCppEmbed, filepath.Join(filepath.Dir(r), "*"))
+		embeddedRunnerDir := filepath.Join(filepath.Dir(r), "*")
+		if runtime.GOOS == "windows" {
+			// in this case we need to convert the filepath to a unix filepath, since that is how go embeds the files
+			embeddedRunnerDir = strings.ReplaceAll(embeddedRunnerDir, "\\", "/")
```

Comment:
This is only necessary because runners are set with `filepath.Join` instead of `path.Join`. Why not just use `path.Join`?